### PR TITLE
JGC-238 fix folder move/copy with parentheses

### PR DESCRIPTION
**Pull request by Postyy**

### CODE REVIEW

File: utils/parenthesesutils.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+			return true
+		}
+	}
+	return false
+}
+
+func RemovePlaceholderParentheses(pattern, target string) string {
+	parentheses := ParenthesesSlice{getParenthesesIndex(pattern, target)}
+
+	//remove parentheses that have an associate placeholder
```
Comment (eyalbe4) [inline utils/parenthesesutils.go]: //remove  --> // Remove parentheses which have a corresponding placeholder.
(notice the upper-case R and the leading white-space).

File: artifactory/services/utils/aqlquerybuilder.go [hunk 2]
```diff
@@ -327,3 +326,14 @@ func getAqlValue(val string) string {
 	}
 	return fmt.Sprintf(aqlValuePattern, val)
 }
+func prepareSourceSearchPattern(pattern, target string, repositoryExists bool) string {
```
[the remark this answers is not in the document]
Comment (Postyy) [reply artifactory/services/utils/aqlquerybuilder.go]: Done.

File: utils/parenthesesutils.go (hunk 1 above)
Comment (Postyy) [reply utils/parenthesesutils.go]: Done.

File: utils/parenthesesutils.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+		}
+	}
+	if placeholderFound != nil {
+		inPlaceRemoveDuplicate(&placeholderFound)
+	}
+	return placeholderFound
+}
+
+// Return the array of each parentheses in pattern with assoiciate place holder in target
+func getParenthesesIndex(pattern, target string) []Parentheses {
```
[the remark this answers is not in the document]
Comment (Postyy) [reply utils/parenthesesutils.go]: You're right about the name of the function, but my suggestion for the description:
`Find all of the parentheses in the pattern, which does correspond to placeholders defined in the target`

**Pull request merged**